### PR TITLE
There is a need to get parent artifact for RootArtifact.

### PR DESCRIPTION
**Pull request by piddubnyi**
Solving https://github.com/jcabi/jcabi-aether/issues/70
Added method that downloads parent.

### CODE REVIEW

**Comment by dmarkov** [comment]
@piddubnyi I will find a reviewer for your pull requests shortly, thanks for contribution!

**Comment by dmarkov** [comment]
@mkordas review this one please

**Comment by mkordas** [comment]
@piddubnyi I'm on it

**Comment by mkordas** [comment]
@piddubnyi first of all, can you fix failing build?

File: src/main/java/com/jcabi/aether/RootArtifact.java [hunk 1]
```diff
@@ -121,6 +129,47 @@ public Artifact artifact() {
...
+     * @return Parent artifact
+     * @throws Exception if downloading fails
+     */
+    public Artifact lookUpParent(final RepositorySystem system,
+        final RepositorySystemSession session,
+        final List<RemoteRepository> repos)
+        throws Exception {
+        Artifact result = null;
+        final Parent parent = new DefaultModelBuilder()
+            .build(new DefaultModelBuildingRequest()
```
Comment (mkordas) [inline src/main/java/com/jcabi/aether/RootArtifact.java]: @piddubnyi there's something wrong with brackets structure here

File: src/main/java/com/jcabi/aether/RootArtifact.java [hunk 2]
```diff
@@ -121,6 +129,47 @@ public Artifact artifact() {
...
+     * @param system The Repository System
+     * @param session The Repository Session
+     * @param repos Repositories to send request
+     * @return Parent artifact
+     * @throws Exception if downloading fails
+     */
+    public Artifact lookUpParent(final RepositorySystem system,
+        final RepositorySystemSession session,
+        final List<RemoteRepository> repos)
+        throws Exception {
```
Comment (mkordas) [inline src/main/java/com/jcabi/aether/RootArtifact.java]: @piddubnyi please move `throws Exception {` to the line above

File: src/main/java/com/jcabi/aether/RootArtifact.java [hunk 3]
```diff
@@ -121,6 +129,47 @@ public Artifact artifact() {
...
+        Artifact result = null;
```
Comment (mkordas) [inline src/main/java/com/jcabi/aether/RootArtifact.java]: @piddubnyi can we avoid `null` here? Please read http://www.yegor256.com/2014/05/13/why-null-is-bad.html

**Comment by mkordas** [comment]
@piddubnyi I don't see any tests for the feature, how do you know that it works?

File: src/main/java/com/jcabi/aether/RootArtifact.java [hunk 4]
```diff
@@ -121,6 +129,47 @@ public Artifact artifact() {
...
+            .getParent();
+        if (parent != null) {
+            result = system.resolveArtifact(
+                session,
+                new ArtifactRequest()
+                    .setArtifact(
+                        new DefaultArtifact(
+                            parent.getGroupId(),
+                            parent.getArtifactId(),
+                            "pom",
```
Comment (mkordas) [inline src/main/java/com/jcabi/aether/RootArtifact.java]: @piddubnyi please apply principle from http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#indentation
> The second important rule of indentation says that you should put as much as possible on one line - within the limit of 80 characters.

File: src/main/java/com/jcabi/aether/RootArtifact.java [hunk 5]
```diff
@@ -121,6 +129,47 @@ public Artifact artifact() {
...
+                            parent.getVersion()
+                        )
+                ).setRepositories(repos)
+            ).getArtifact();
```
Comment (mkordas) [inline src/main/java/com/jcabi/aether/RootArtifact.java]: @piddubnyi please format fluent calls according to http://www.yegor256.com/2014/10/23/paired-brackets-notation.html#fluent

File: src/main/java/com/jcabi/aether/RootArtifact.java [hunk 6]
```diff
@@ -121,6 +129,47 @@ public Artifact artifact() {
     }
 
     /**
+     * Downloads parent artifact from repositories.
```
Comment (mkordas) [inline src/main/java/com/jcabi/aether/RootArtifact.java]: @piddubnyi can you explain this concept bit more here in Javadoc?

**Comment by mkordas** [comment]
@piddubnyi please merge or rebase over master so build for Java 6 will not be completely failed

**Comment by mkordas** [comment]
@piddubnyi ping

**Comment by mkordas** [comment]
@piddubnyi do you need help here?

**Comment by mkordas** [comment]
@piddubnyi are you going to finish this PR?

**Comment by mkordas** [comment]
@piddubnyi ping

**Comment by mkordas** [comment]
@piddubnyi do you have time to continue here?

**Comment by mkordas** [comment]
@piddubnyi you are assigned to this task for more than 10 days, you need to take some action here

**Comment by mkordas** [comment]
@piddubnyi ping

**Comment by piddubnyi** [comment]
@mkordas Sorry, I'm leaving this one.

**Comment by mkordas** [comment]
@piddubnyi that's OK to drop PR, no worries. Thanks for letting me know!

**Comment by dmarkov** [comment]
@mkordas paid, thanks, added **33 mins** to your account, payment `56b9f01b94e5fa626f0000a3`, 269 hours and 2 mins was spent. you're getting extra minutes here (c=18). added +33 to your rating, now it is equal to [+5530](http://www.netbout.com/b/36080?open=rating)

**Pull request closed without merging**